frontend/javascripts/oxalis/model/sagas/min_cut_saga.js

```diff
@@ -0,0 +1,694 @@
+  for (const [resolutionIndex, targetMag] of appropriateResolutionInfos) {
+    try {
+      console.log("Trying min-cut computation at", targetMag);
+      console.group();
+      yield* call(
+        tryMinCutAtMag,
+        targetMag,
+        resolutionIndex,
+        boundingBoxMag1,
+        nodes,
+        volumeTracingLayer,
+      );
+      console.groupEnd();
+
+      for (
+        let refiningResolutionIndex = resolutionIndex - 1;
+        refiningResolutionIndex >= 0;
+        refiningResolutionIndex--
+      ) {
+        // Refine min-cut on lower resolutions, if they exist.
+        if (!resolutionInfo.hasIndex(refiningResolutionIndex)) {
+          continue;
+        }
+
+        const refiningResolution = resolutionInfo.getResolutionByIndexOrThrow(
+          refiningResolutionIndex,
+        );
+        console.log("Refining min-cut at", refiningResolution);
+        console.group();
+        yield* call(
+          tryMinCutAtMag,
+          refiningResolution,
+          refiningResolutionIndex,
+          boundingBoxMag1,
+          nodes,
+          volumeTracingLayer,
+        );
+        console.groupEnd();
+      }
+
+      return;
+    } catch (exception) {
+      if (exception === TimeoutError) {
+        console.log("Retrying at higher mag if possible...");
+      } else if (exception === PartitionFailedError) {
+        console.warn(exception);
+        return;
+      } else {
+        throw exception;
+      }
+    }
+  }
+  console.warn("Couldn't perform min-cut due to timeout");
+}
+
+//
+// Algorithmic implementation of the min cut approach.
+//
+// The directed (!) voxel graph is defined so that each voxel is a node and
+// two nodes are connected if they are neighbors and have the same segment id.
+// Since the graph is directed, all edges are initially symmetric double-edges.
+//
+// The algorithm looks for shortest paths between two given seeds A and B (via
+// breadth-first searches).
+// In each iteration, a shortest path between A and B is removed until no paths
+// exist anymore. Then, the two seeds are separated from each other.
+//
+// When removing a path, only the edges are removed which point from A to B.
+// This leaves "back-edges" in the graph (also known as "residuals").
+// In the final phase, these residuals are traversed to find out which nodes
+// cannot be reached, anymore. These nodes are the one that should be erased
+// to separate A from B.
+
+function* tryMinCutAtMag(
+  targetMag,
+  resolutionIndex,
+  boundingBoxMag1,
+  nodes,
+  volumeTracingLayer,
+): Saga<void> {
+  console.log("resolutionIndex, targetMag", { resolutionIndex, targetMag });
+
+  const boundingBoxTarget = boundingBoxMag1.fromMag1ToMag(targetMag);
+
+  const globalSeedA = V3.fromMag1ToMag(nodes[0].position, targetMag);
+  const globalSeedB = V3.fromMag1ToMag(nodes[1].position, targetMag);
+
+  let minDistToSeed = MIN_DIST_TO_SEED / targetMag[0];
```

Comment:
this is not correct, since the three dimensions have to be taken into account each.